tools/purge_omsagent.sh

```diff
@@ -0,0 +1,407 @@
+            INSTALLER="RPM"
+        fi
+    fi
+
+    # Fall back on detection via package manager availability
+    if [ "$INSTALLER" = "" ]; then
+        if [ -x "$(command -v dpkg)" ]; then
+            INSTALLER="DPKG"
+        elif [ -x "$(command -v rpm)" ]; then
+            INSTALLER="RPM"
+        fi
+    fi
+
+    if [ "$INSTALLER" = "" ]; then
+        echo "Error: This system does not have a supported package manager" >&2
+        echo "Supported Sytems: 'DPKG' & 'RPM'" >&2
+        echo "Cannot purge without a supported package manager" >&2
+        call_exit $UNSUPPORTED_PKG_INSTALLER
+    fi
+}
+
+ulinux_detect_servicecontroller()
+{
+    CONTROLLER=""
+
+    # Detect based on systemd
+    if which systemctl > /dev/null 2>&1; then
+        CONTROLLER="systemctl"
+    
+    # Fall back on detection via filepath
+    else
+        if   [ -f "/bin/systemctl" ]; then
+            CONTROLLER="systemctl"
+        elif [ -f "/sbin/service" ]; then
+            CONTROLLER="service"
+        elif [ -f "/usr/sbin/invoke-rc.d" ]; then
+            CONTROLLER="invoke-rc.d"
+        fi
+    fi
+
+    if [ "$CONTROLLER" = "" ]; then
+        echo "Error: This system does not have a supported service controller" >&2
+        echo "Supported Systems: 'systemctl', 'service', and 'invoke-rc'" >&2
+        echo "Cannot purge without a supported service controller" >&2
+        call_exit $UNSUPPORTED_SERVICE_CONTROLLER
+    fi
+}
+
+# $1 - The name of the package to check as to whether it's installed
+check_if_pkg_is_installed()
+{
+    if [ "$INSTALLER" = "DPKG" ]; then
+        dpkg -s $1 2> /dev/null | grep Status | grep " installed" 1> /dev/null
+    else
+        rpm -q $1 > /dev/null 2>&1
+    fi
+
+    return $?
+}
+
+# $1 - The name of the package to be uninstalled
+pkg_rm()
+{
+    if [ "$1" != "" ]; then
+        echo "----- Removing package: $1 -----"
+        if [ "$INSTALLER" = "DPKG" ]; then
+            dpkg --purge ${1}
+        else
+            rpm --erase ${1}
+        fi
+        if [ $? -ne 0 ]; then
+            echo "----- Ignore previous errors for package: $1 -----"
+        fi
+    fi
+}
+
+check_service_running()
+{
+    if [ "$CONTROLLER" = "systemctl" ]; then
+        systemctl status $WAAGENT_NAME 2> /dev/null | grep Active | grep " active (running)" 1> /dev/null
+    elif [ "$CONTROLLER" = "service" ]; then
+        service $WAAGENT_NAME status 2> /dev/null | grep $1 | grep "is running..." 1> /dev/null
+    else
+        invoke-rc.d $WAAGENT_NAME status 2> /dev/null | grep $1 | grep "start/running" 1> /dev/null
+    fi
+
+    return $?
+}
+
+check_waagent_name()
+{
+    WAAGENT_NAME=""
+
+    if [ "$CONTROLLER" = "systemctl" ]; then
+        if systemctl status waagent > /dev/null 2>&1; then
+            WAAGENT_NAME="waagent"
+        elif systemctl status walinuxagent > /dev/null 2>&1; then
+            WAAGENT_NAME="walinuxagent"
+        fi
+    elif [ "$CONTROLLER" = "service" ]; then
```

Comment:
nit
can make second 2 cases into 1 else and use `$CONTROLLER waagent status`
same for other functions

Reply:
Fixed in this commit: https://github.com/microsoft/OMS-Agent-for-Linux/pull/1255/commits/6084ab38cd50be252158f02bd813a2b094ffb573